### DISCUSSION TITLE
Codespaces port forwarding not working (mainly for flask app)?

### DISCUSSION THREAD

**Discussion by dharantej1** [community/community#41907] [Codespaces] [answered]
### Codespaces Port Forwarding issue

Question

### Body

I was doing my flask app for a Deep Learning project and i have chosen codespaces when i knew about it. But a week before when i started executing my flask app again, the port forwarding was not happening and i am unable to run the frontend (in my case html page)

is this the issue only for me, or anyone else got the same issue

if you have found whats the problem, please help me out ASAP!

you can check my application here https://github.com/dharantej1/KMIT_IOMP and please tell me whats the issue?

**Comment by erikaellr** [comment, marked as answer]
Hi @dharantej1, we're thrilled you're using Codespaces! As you experienced at first, Codespaces and VSCode attempt to auto-detect any services you are running in the codespace and have VSCode forward them, but if the service you're running is not being auto-detected, you can forward the ports manually after starting the service, or you can set up a `devcontainer.json` to specify which ports should be forwarded every time you connect to a codespace for that repo. See [the docs on forwarding ports in Codespaces](https://docs.github.com/en/codespaces/developing-in-codespaces/forwarding-ports-in-your-codespace) for more detail. Hope this helps, and good luck!
